storage/storage.cpp

```diff
@@ -1145,4 +1145,19 @@ void Storage::DoClickOnDownloadMap(TCountryId const & countryId)
   if (m_downloadMapOnTheMap)
     m_downloadMapOnTheMap(countryId);
 }
+
+void Storage::ForEachInSubtree(TCountryId const & root, TForEachFunction const & toDo) const
```

Comment:
one more sample:

void Process(TFn const & fn)
{
    fn();
}

class MyFn
{
public:
    bool called = false;
    void Do() { called = true; }
};

int main()
{
    MyFn m1, m2;
    Process(bind(&MyFn::Do, ref(m1)));
    assert(m1.called == true);
    Process(bind(&MyFn::Do, &m2));
    assert(m2.called == true);
}

'TFn const & fn' param allows to change state of MyFn, but fn itself cannot be changed.